californium-core/src/main/java/org/eclipse/californium/core/CoapClient.java

```diff
@@ -930,7 +927,67 @@ protected Endpoint getEffectiveEndpoint(Request request) {
 			return EndpointManager.getEndpointManager().getDefaultEndpoint();
 		}
 	}
+
+	/*
+	 * Create a GET request with a type specified in the client
+	 *
+	 * @return the request
+	 */
+	private Request newGet() {
+		return applyRequestType(Request.newGet());
+	}
+
+	/*
+	 * Create a POST request with a type specified in the client
+	 *
+	 * @return the request
+	 */
+	private Request newPost() {
+		return applyRequestType(Request.newPost());
+	}
+
+	/*
+	 * Create a PUT request with a type specified in the client
+	 *
+	 * @return the request
+	 */
+	private Request newPut() {
+		return applyRequestType(Request.newPut());
+	}
+
+	/*
+	 * Create a DELETE request with a type specified in the client
+	 *
+	 * @return the request
+	 */
+	private Request newDelete() {
+		return applyRequestType(Request.newDelete());
+	}
+
+	/*
+	 * Applies the CoapClient#type (managed by useNONs, useCONs) to the specified request.
+	 *
+	 *
+	 * @param request the request
+	 * @return the same request with changed type
+	 */
+	private Request applyRequestType(Request request) {
+		request.setType(this.type);
+		return request;
+	}
 	
+	/*
+	 * Assigns a CoapClient#uri if request has no uri.
+	 *
+	 * @param request the request
+	 */
+	private void assignClientUriIfEmpty(Request request) {
+		// request.getUri() is a computed getter and never returns null so checking options
+		if (request.getOptions().getUriHost() == null) {
```

Comment:
At least the *destination* property seems to be always set when someone sets the URI on the request. So, yes, I think it would be a better choice for the check ...